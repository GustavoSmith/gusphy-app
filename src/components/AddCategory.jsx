import { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const onSubmit = (e) => {
    e.preventDefault();
    const cleanValue = inputValue.trim();
    if (cleanValue.length < 2) return;
    onNewCategory(cleanValue);
    setInputValue("");
    localStorage.setItem("lastCategory", cleanValue);
  };

  return (
    <form onSubmit={onSubmit} className="flex flex-col items-center gap-2">
      <input
        type="text"
        className="rounded-md border-2 border-slate-600 p-2 md:mb-6 md:w-[400px] xl:w-[600px]"
        placeholder="Buscar Gifs"
        value={inputValue}
        onChange={(e) => setInputValue(e.target.value)}
      />
      <button
        className="rounded-full bg-yellow-300 px-4 py-2 font-semibold text-black"
        type="submit"
      >
        Buscar
      </button>
    </form>
  );
};
