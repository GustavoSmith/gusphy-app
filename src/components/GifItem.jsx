import PropTypes from "prop-types";
import Swal from "sweetalert2";

export const GifItem = ({ title, url }) => {
  const handleCopyToClipboard = () => {
    navigator.clipboard.writeText(url);

    Swal.fire({
      toast: true,
      icon: "success",
      title: "Copiado al portapapeles",
      position: "top-end",
      customClass: {
        timerProgressBar: "bg-blue-300",
      },
      showConfirmButton: false,
      timer: 2000,
      timerProgressBar: true,
      didOpen: (toast) => {
        toast.addEventListener("mouseenter", Swal.stopTimer);
        toast.addEventListener("mouseleave", Swal.resumeTimer);
      },
    });
  };

  return (
    <button
      onClick={handleCopyToClipboard}
      className="flex flex-col content-center items-center justify-center overflow-hidden rounded-xl bg-white shadow-md md:w-2/5 lg:w-1/4"
    >
      <img src={url} alt={title} />
      <p className="m-4 bg-white text-center text-lg">{title}</p>
    </button>
  );
};

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
