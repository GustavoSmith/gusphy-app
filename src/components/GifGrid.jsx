import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks";

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h2 className="my-8 text-4xl font-semibold">{category}</h2>

      {isLoading && <p className="text-3xl">Cargando...</p>}

      <div className="flex flex-wrap justify-center gap-6">
        {images.map((image) => (
          <GifItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};
