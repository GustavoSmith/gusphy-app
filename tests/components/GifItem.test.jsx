import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Pruebas en <GifItem />", () => {
  const title = "Un título";
  const url = "https://localhost/ejemplo.jpg";
  test("Debe matchear la snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("Debe mostrar la imagen con el URL y con su 'alt' correspondiente ", () => {
    render(<GifItem title={title} url={url} />);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("Debe mostrar el título", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
