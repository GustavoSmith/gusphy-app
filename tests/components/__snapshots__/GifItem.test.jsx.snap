// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe matchear la snapshot 1`] = `
<div>
  <button
    class="flex flex-col content-center items-center justify-center overflow-hidden rounded-xl bg-white shadow-md md:w-2/5 lg:w-1/4"
  >
    <img
      alt="Un título"
      src="https://localhost/ejemplo.jpg"
    />
    <p
      class="m-4 bg-white text-center text-lg"
    >
      Un título
    </p>
  </button>
</div>
`;
